nautobot/extras/tests/test_jobs.py

```diff
@@ -386,6 +389,108 @@ def test_run_job_fail(self):
             self.assertEqual(FileProxy.objects.count(), 0)
 
 
+@skipIf("dummy_plugin" not in settings.PLUGINS, "dummy_plugin not in settings.PLUGINS")
+class JobSingletonTestCase(CeleryTestCase):
+    """Test cases for Jobs with `singleton=True` in their `Meta`."""
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+    def test_already_queued_no_arguments(self):
+        """Executing the same task IGNORING args results in an `AlreadyQueued` error."""
+        from dummy_plugin.tasks import slow_add  # noqa: Avoid import issues
+
+        slow_add.apply_async(args=(1, 3), kwargs={"interval": 0.1}, singleton=True)
+        with self.assertRaises(AlreadyQueued):
+            slow_add.apply_async(args=(1, 2), kwargs={"interval": 0.1}, singleton=True)
+
+    def test_already_queued_with_arguments(self):
+        """Executing the same task INCLUDING args results in an `AlreadyQueued` error."""
+        from dummy_plugin.tasks import slow_add  # noqa: Avoid import issues
+
+        slow_add.apply_async(args=(1, 2), kwargs={"interval": 0.1}, singleton=True, once={"keys": ["a", "b"]})
+        with self.assertRaises(AlreadyQueued):
+            slow_add.apply_async(args=(1, 2), kwargs={"interval": 0.1}, singleton=True, once={"keys": ["a", "b"]})
+
+    def test_celery_kwargs(self):
+        # TODO(jathan): Implement test for celery_kwargs passthru
+        assert True
+
+    def test_singleton_job_no_arguments(self):
+        """Singleton Job execution WITHOUT arguments results in an `AlreadyQueued` error."""
+        with self.settings(JOBS_ROOT=os.path.join(settings.BASE_DIR, "extras/tests/dummy_jobs")):
+            job_name = "local/test_singleton/SingletonJobNoArguments"
```

Comment:
Can we change "arguments"/"no arguments" to "singleton_keys"/"no singleton_keys" throughout? I think it'd be clearer.

Reply:
Yes, that is fair.